python-threatexchange/threatexchange/exchanges/impl/ncmec_api.py

```diff
@@ -138,18 +138,29 @@ class NCMECSignalExchangeAPI(
         3. As of 5/2022 there are no false positive or seen statuses
     """
 
-    MAX_FETCH_SIZE = 400000
-    FETCH_SHRINK_FACTOR = 4
+    MAX_FETCH_SIZE: t.ClassVar[int] = 400000
+    FETCH_SHRINK_FACTOR: t.ClassVar[int] = 4
 
     def __init__(
         self,
-        username: str = "",
-        password: str = "",
+        collab: NCMECCollabConfig,
+        username: str,
+        password: str,
     ) -> None:
         super().__init__()
+        self.collab = collab
         self._username = username
         self._password = password
 
+    @classmethod
+    def for_collab(
+        cls,
+        collab: NCMECCollabConfig,
+        credentials: t.Optional["NCMECCredentials"] = None,
```

Comment:
Yay! Backwards compatible.

Reply:
I think we arrived at a solution that lets HMA add support for editable credentials gradually. I like this.